homeassistant/components/immich/media_source.py

```diff
@@ -39,13 +38,14 @@ class ImmichMediaSourceIdentifier:
 
     def __init__(self, identifier: str) -> None:
         """Split identifier into parts."""
-        parts = identifier.split("/")
-        # config_entry.unique_id/collection/collection_id/asset_id/file_name
+        parts = identifier.split("|")
+        # config_entry.unique_id|collection|collection_id|asset_id|file_name|mime_type
         self.unique_id = parts[0]
         self.collection = parts[1] if len(parts) > 1 else None
         self.collection_id = parts[2] if len(parts) > 2 else None
         self.asset_id = parts[3] if len(parts) > 3 else None
         self.file_name = parts[4] if len(parts) > 3 else None
+        self.mime_type = parts[5] if len(parts) > 3 else None
```

Comment:
shouldn't those be 4 and 5 respectively?
```suggestion
        self.file_name = parts[4] if len(parts) > 4 else None
        self.mime_type = parts[5] if len(parts) > 5 else None
```

Reply:
I asked copilot if we could make this nicer
```
parts = identifier.split("|")
(
    self.unique_id,
    self.collection,
    self.collection_id,
    self.asset_id,
    self.file_name,
    self.mime_type,
) = (parts + [None] * 6)[:6]
```

Reply:
> I asked copilot if we could make this nicer

I know why I've mixed feelings with copilot 🙈

Reply:
> I know why I've mixed feelings with copilot 🙈

Me too, but sometimes it's really helpful/helps tidying up